umbra-js/src/classes/KeyPair.ts

```diff
@@ -92,16 +91,14 @@ export class KeyPair {
       throw new Error('Must provide instance of RandomNumber');
     }
     // Get shared secret to use as encryption key
-    const ephemeralWallet = Wallet.createRandom();
-    const sharedSecret = getSharedSecret(ephemeralWallet.privateKey, this.publicKeyHex);
+    const ephemeralPrivateKey = hexlify(utils.randomPrivateKey()); // private key as hex with 0x prefix
+    const ephemeralPublicKey = `0x${getPublicKey(ephemeralPrivateKey.slice(2))}`; // public key as hex with 0x prefix
```

Comment:
Why use hexlify in the line above but string interpolation here? Is it doing the same thing?

Reply:
Hexlify is taking in a Uint8Array of bytes in that case and converting it to a hex string